### DISCUSSION TITLE
Return null or Empty result for requests?

### DISCUSSION THREAD

**Discussion by tziemek** [artiso-solutions/adsd-hotel#30] [General] [answered]
In our black we have a method to search for the guest information for a specific order id or find the order id for a given name.
What should we return if no matching entry was found? The general way would be to return `null` but with the late emerging of NonNullableReferenceTypes I lean more toward returning an empty (static) result.

**Comment by tommasobertoni** [comment]
Given that the nullability is enabled in all the projects, you'd get a warning/error on the consumer side when you don't check for `null`: I'm good with that.

```csharp
GuestInformation? guestInfo = await ambassador.GetGuestInfo(orderId);
if (guestInfo is null) { }
else { }
```

**Comment by tziemek** [comment, marked as answer]
We agreed on putting exception information in the response type, maybe use a base class for that.
